packages/core/http/core-http-server-internal/src/lifecycle_handlers.ts

```diff
@@ -61,12 +61,18 @@ export const createVersionCheckPostAuthHandler = (kibanaVersion: string): OnPost
 };
 
 export const createCustomHeadersPreResponseHandler = (config: HttpConfig): OnPreResponseHandler => {
-  const { name: serverName, securityResponseHeaders, customResponseHeaders } = config;
+  const {
+    name: serverName,
+    securityResponseHeaders,
+    customResponseHeaders,
+    csp: { header: cspHeader },
+  } = config;
 
   return (request, response, toolkit) => {
     const additionalHeaders = {
       ...securityResponseHeaders,
       ...customResponseHeaders,
+      'Content-Security-Policy': cspHeader,
       [KIBANA_NAME_HEADER]: serverName,
     };
```

Comment:
The approach seems good and to work fine with the usages we're currently having of CSP. My main concern is that I wonder if we're not going in a direction that would totally block us for https://github.com/elastic/kibana/issues/93785 later? 

WDYT?

Reply:
Would using a [CSP hash](https://content-security-policy.com/hash/) solve the problem. That way we could hardcode it at build time (instead of having it being random for each request) and then we could just return it for all requests.

Reply:
@pgayvallet What's the reasoning behind wanting to inline `bootstrap.js`?

Reply:
> Would using a [CSP hash](https://content-security-policy.com/hash/) solve the problem

Don't think it does for some planned usages, e.g https://github.com/elastic/kibana/issues/132595

> What's the reasoning behind wanting to inline bootstrap.js?

Perf, avoiding a roundtrip can could be avoided.

But https://github.com/elastic/kibana/issues/132595 is another (more concrete) one that could be blocked by this PR's design proposal.

More globally, I think we do need a proper understanding on how we're planning to allow per-request enhancing of the CSP header before doing any change to the current (flawed) implementation, to avoid blocking us later.

Reply:
cc @lukeelmers for visibility

Reply:
I'm currently leaning towards ignoring the need for CSP nonce/hash in this PR. I think it's easier to simply merge this as is and then, once we get to it, refactor to support the new use-cases. It's not any more work to do it now than to do it later, if anything, it's easier to implement once we have a concrete use-case.

Reply:
Thanks for linking the "sandboxed iframe" issue @pgayvallet, I can't find any feature mentioned that requires such an API but that's not to say there isn't one. 

I agree with Thomas here, given that we currently don't inline `bootstrap.js` and are not making anything worse as part of this PR, it shouldn't be a blocker that we might require CSP nonce/hash in the future if we ever want to inline it.